integration/netmon/netmon_test.bats

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../lib/common.bash"
+
+# Environment variables
+IMAGE="busybox"
+NETWORK_NAME="test"
+CONTAINER_NAME="containerA"
+PAYLOAD_ARGS="tail -f /dev/null"
+
+setup() {
+	clean_env
+
+	# Check that processes are not running
+	run check_processes
+	echo "$output"
+	[ "$status" -eq 0 ]
+
+	extract_kata_env
+
+	# Enabling netmon at the configuration file
+	sudo sed -i -e 's/^#\(enable_netmon =\).*$/\1 true/g' ${RUNTIME_CONFIG_PATH}
+}
+
+@test "test netmon" {
+	# Create network
+        docker network create $NETWORK_NAME
+
+	# Run container
+	docker run --runtime=$RUNTIME -d --name $CONTAINER_NAME $IMAGE $PAYLOAD_ARGS
+
+	# Check the number of interfaces before network connect
+	before_interfaces=$(docker exec $CONTAINER_NAME ip addr show | awk '/inet.*brd/{print $NF}' | wc -l)
+
+	docker network connect $NETWORK_NAME $CONTAINER_NAME
+
+	sleep 15
```

Comment:
do we still need this sleep?

Reply:
Really we should make it a sleepy poll loop with a long timeout - so, poll every 5s, and if after 60s or something the IP has not appeared, then fail. Relying on a hard-wired sleep is going to fail one day when we change some CI infra somewhere...